packages/graph-explorer/src/connector/sparql/templates/keywordSearch/keywordSearchBlankNodesIdsTemplate.ts

```diff
@@ -31,7 +31,7 @@ export default function keywordSearchBlankNodesIdsTemplate({
             ${getSubjectClasses(subjectClasses)}
             ${getFilterObject(exactMatch, searchTerm)}
         }
-        ${limit > 0 ? `LIMIT ${limit} OFFSET ${offset}` : ""}
```

Comment:
Is limit value user-provided? If yes would it make sense to keep the greater than zero check here?

Reply:
I see this definition of KeywordSearchRequest where it says if limit is zero then that means no limit. If limit is zero here then it will send the request with `LIMIT 0` which might actually return zero results unless the server also interprets LIMIT 0 as unlimited.

```
export type KeywordSearchRequest = {
  /**
   * Search term to match with vertices attributes
   */
  searchTerm?: string;
  /**
   * Filter by attribute names.
   */
  searchByAttributes?: Array<string>;
  /**
   * Filter by vertex types.
   */
  vertexTypes?: Array<string>;
  /**
   * Limit the number of results.
   * 0 = No limit.
   */
  limit?: number;
  /**
   * Skip the given number of results.
   */
  offset?: number;
  /**
   * Only return exact attribute value matches.
   */
  exactMatch?: boolean;
};
```

Reply:
This particular limit is not a user input, but I like the idea of keeping the `0 == no limit` behavior. I'll make the change.

Reply:
I added the logic to the tests. Since JavaScript treats zero as a false value, none of the logic needed to change.